master/buildbot/db/logs.py

```diff
@@ -427,7 +427,7 @@ def thddeleteOldLogs(conn):
 
         return self.db.pool.do(thddeleteOldLogs)
 
-    def _logdictFromRow(self, row):
-        rv = dict(row)
+    def _logdictFromRow(self, row: sa.engine.row.Row):
+        rv = dict(row._mapping)  # noqa pylint: disable=protected-access
```

Comment:
Using internal variable doesn't seem like a long-term solution. I think we should use mappings() function https://docs.sqlalchemy.org/en/14/core/connections.html#sqlalchemy.engine.Result.mappings

Reply:
This is indeed far better,
I guess I use the `_mapping` as it lead to less change, but that's a poor reason and `mappings()` is a far better alternative.
I'll update the PR.